src/main/java/com/fouristhenumber/utilitiesinexcess/mixins/early/minecraft/MixinMobSpawnerBaseLogic_CursedEarthSpawner.java

```diff
@@ -0,0 +1,105 @@
+package com.fouristhenumber.utilitiesinexcess.mixins.early.minecraft;
+
+import java.util.List;
+
+import net.minecraft.block.Block;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.EntityLiving;
+import net.minecraft.entity.monster.IMob;
+import net.minecraft.potion.Potion;
+import net.minecraft.potion.PotionEffect;
+import net.minecraft.tileentity.MobSpawnerBaseLogic;
+import net.minecraft.util.WeightedRandom;
+import net.minecraft.world.World;
+
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+import com.fouristhenumber.utilitiesinexcess.common.blocks.BlockCursedEarth;
+
+@Mixin(MobSpawnerBaseLogic.class)
+public class MixinMobSpawnerBaseLogic_CursedEarthSpawner {
+
+    @Shadow
+    private int minSpawnDelay;
+    @Shadow
+    private int maxSpawnDelay;
+    @Shadow
+    private List<WeightedRandom.Item> potentialEntitySpawns;
+
+    // If the block below is cursed earth,
+    // always run the spawner instead of depending
+    // on player location.
+    @Inject(method = "isActivated", at = @At("HEAD"), cancellable = true)
+    private void uei$onIsActivated(CallbackInfoReturnable<Boolean> cir) {
+        MobSpawnerBaseLogic self = (MobSpawnerBaseLogic) (Object) this;
+        World world = self.getSpawnerWorld();
```

Comment:
simply call `this.getSpawnerWorld();` and add `    @Shadow
public abstract World getSpawnerWorld();` at the top in your mixin